src/main/java/org/tailormap/api/controller/admin/SolrAdminController.java

```diff
@@ -158,57 +171,110 @@ public ResponseEntity<?> pingSolr() {
       path = "${tailormap-api.admin.base-path}/index/{searchIndexId}",
       produces = MediaType.APPLICATION_JSON_VALUE)
   public ResponseEntity<?> index(@PathVariable Long searchIndexId) {
+    SearchIndex searchIndex = validateInputAndFindIndex(searchIndexId);
+
+    if (searchIndex.getStatus() == SearchIndex.Status.INDEXING) {
+      throw new ResponseStatusException(
+          HttpStatus.CONFLICT,
+          "Indexing already in progress, check tasks overview before retrying");
+    }
+
+    boolean createNewIndex =
+        (null == searchIndex.getLastIndexed()
+            || searchIndex.getStatus() == SearchIndex.Status.INITIAL);
+
+    boolean hasSchedule =
+        (null != searchIndex.getSchedule() && null != searchIndex.getSchedule().getUuid());
+
+    try {
+      if (hasSchedule) {
+        startScheduledJobIndexing(searchIndex);
+      } else {
+        startOneTimeJobIndexing(searchIndex);
+      }
+      searchIndexRepository.save(searchIndex);
+    } catch (UnsupportedOperationException
+        | IOException
+        | SolrServerException
+        | SolrException
+        | SchedulerException e) {
+      logger.error("Error indexing", e);
+      searchIndex.setStatus(SearchIndex.Status.ERROR);
+      searchIndexRepository.save(searchIndex);
+      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, e.getMessage());
+    }
+
+    if (createNewIndex) {
+      logger.info("Started creation of new index for search index {}", searchIndex.getName());
+      return ResponseEntity.status(HttpStatus.ACCEPTED).build();
+    } else {
+      logger.info("Started update of index for search index {}", searchIndex.getName());
+      return ResponseEntity.accepted().build();
+    }
+  }
 
+  private void startOneTimeJobIndexing(SearchIndex searchIndex)
+      throws SolrServerException, IOException, SchedulerException {
+    String taskName =
+        taskManagerService.createTask(
+            IndexTask.class,
+            new TMJobDataMap(
+                Map.of(
+                    Task.TYPE_KEY,
+                    TaskType.INDEX,
+                    Task.DESCRIPTION_KEY,
+                    "One-time indexing of " + searchIndex.getName(),
+                    IndexTask.INDEX_KEY,
+                    searchIndex.getId().toString(),
+                    Task.PRIORITY_KEY,
+                    0)));
+    logger.info("One-time indexing job with UUID {} started", taskName);
```

Comment:
It would be useful to return this UUID in the API response, so the frontend can monitor the specific task created.